src/Features/Lsif/Generator/Program.cs

```diff
@@ -127,30 +136,37 @@ private static async Task LocateAndRegisterMSBuild(TextWriter logFile)
             MSBuildLocator.RegisterInstance(msbuildInstance);
         }
 
-        private static async Task GenerateFromProjectAsync(FileInfo projectFile, ILsifJsonWriter lsifWriter, TextWriter logFile)
+        private static async Task GenerateFromProjectAsync(
+            FileInfo projectFile, ILsifJsonWriter lsifWriter, TextWriter logFile, CancellationToken cancellationToken)
         {
             await GenerateWithMSBuildWorkspaceAsync(
                 projectFile, lsifWriter, logFile,
                 async w =>
                 {
                     var project = await w.OpenProjectAsync(projectFile.FullName);
                     return project.Solution;
-                });
+                },
+                cancellationToken);
         }
 
-        private static async Task GenerateFromSolutionAsync(FileInfo solutionFile, ILsifJsonWriter lsifWriter, TextWriter logFile)
+        private static async Task GenerateFromSolutionAsync(
+            FileInfo solutionFile, ILsifJsonWriter lsifWriter, TextWriter logFile, CancellationToken cancellationToken)
         {
             await GenerateWithMSBuildWorkspaceAsync(
                 solutionFile, lsifWriter, logFile,
-                w => w.OpenSolutionAsync(solutionFile.FullName));
+                w => w.OpenSolutionAsync(solutionFile.FullName),
+                cancellationToken);
         }
 
         // This method can't be loaded until we've registered MSBuild with MSBuildLocator, as otherwise
         // we load ILogger prematurely which breaks MSBuildLocator.
         [MethodImpl(MethodImplOptions.NoInlining)]
         private static async Task GenerateWithMSBuildWorkspaceAsync(
-            FileInfo solutionOrProjectFile, ILsifJsonWriter lsifWriter, TextWriter logFile,
-            Func<MSBuildWorkspace, Task<Solution>> openAsync)
+            FileInfo solutionOrProjectFile,
+            ILsifJsonWriter lsifWriter,
+            TextWriter logFile,
+            Func<MSBuildWorkspace, Task<Solution>> openAsync,
```

Comment:
Should this be passing through cancellation tokens?

```suggestion
            Func<MSBuildWorkspace, CancellationToken, Task<Solution>> openAsync,
```

Reply:
done.